specs/vep-2420-getting-started-with-my-data/README.md

```diff
@@ -0,0 +1,198 @@
+* **Status:** draft
+
+
+- [VEP-2420: VDK Getting Started with My Data](#vep-2420-getting-started-with-my-data)
+  - [Summary](#summary)
+  - [Glossary](#glossary)
+  - [Motivation](#motivation)
+    - [Poor assistance around configuration options](#Poor-assistance-around-configuration-options)
+    - [Layers of abstraction aren't clear](#Layers-of-abstraction-are-not-clear)
+    - [Poor documentation around secrets/properties](#Poor-documentation-around-secrets-and-properties)
+    - [Not trivial to run in the IDE](#Not-trivial-to-run-in-the-IDE)
+    - [Reliant on env variables](#Reliant-on-env-variables)
+  - [Requirements and goals](#requirements-and-goals)
+  - [High-level design](#high-level-design)
+  - [API design](#api-design)
+  - [Detailed design](#detailed-design)
+  - [Implementation stories](#implementation-stories)
+  - [Alternatives](#alternatives)
+
+## Summary
+
+Users struggle to get started with their own data in VDK.
+When evaluating a data platform like VDK one of the first actions is often to connect it to your own data sources and play around with it.
+Given that this is not a trivial task on VDK it most likely turns a lot of prospective users off.
+The issues range from:
+1. Poor assistance around configuration options
+2. Layers of abstraction aren't clear
+3. Poor documentation around secrets/properties
+4. Not trivial to RUN in IDE
+5. Reliant on env variables
+
+## Glossary
+
+## Motivation
+At the moment, VKD users face multiple challenges related to using their own data in VDK.
+Based on user feedback we've received, these can be split into 5 categories.
+
+### Poor assistance around configuration options
+
+VDK documents all config and all config can be seen with 'vdk config-help'.
+However despite this there are a number of issues which make it difficult to configure VDK.
+
+
+1. Users are expected to configure their job using a config.ini file. Ini files are not popular in the industry
+2. Sections are confusing
+ini files support sections (https://en.wikipedia.org/wiki/INI_file#Sections).
+Within VDK there are a few different sections, [vdk],[job], [owner].
+Newer properties always go in the [vdk] section.
+It's not clear which properties go in which section. In fact some properties can exist in different sections and will have the same result.
+3. There is no validation on the file at run time to make sure all properties exist. Ideally the system should throw an error if there are properties in the file which don't actually exist.
+4. There is no IDE assistance. Frameworks like spring support autocomplete in their property and yaml files.
+
+### Not obvious what properties do
+Below are the properties most jobs need to set.
+```bash
+export VDK_DB_DEFAULT_TYPE=SQLITE
+export VDK_INGEST_METHOD_DEFAULT=sqlite
+export VDK_INGEST_TARGET_DEFAULT=vdk-sqlite.db
+export VDK_SQLITE_FILE=vdk-sqlite.db
+```
+
+Ingest is a confusing term.
```

Comment:
ingest is not the confusing part. 
Ingest is the process of transferring data from source to destination .

I think the confusing part is " Is this the source" or the "destination"  conifguraoitn ? How do I connect one to the other?  

The title of the issue is "Not obvious what properties do" 

But here we are referring to ingestion configuration. So is this general problem or specific. If it's general why and where. 

I think we have both general problem with properties not easy to see what they do (that's a general problem in the industry but still) 

But also specific problem which partily steps of not having concepts for source and perhaps connection

Reply:
This is a specific problem and I wouldn't normally consider that one property being badly named justifies a section in a VEP. However this is one of the most common properties. It exists across many tutorials and as such needs to be addressed. 
In fixing it maybe we can take learnings from it. 

But I will bring it up at the meeting on Tuesday and we can discuss.

Reply:
I also think ingest is confusing because it assumes there is a primary database that you are ingesting stuff into. 

I think that pipelines should think about moving data from point a to point b and doing a transformation along the way. 

They shouldn't have the concept of a primary database. 
For different teams different databases should be the primary, so no database should be considered primary.

Reply:
Addressed, please resolve if you are happy.